mod/05_standard/set/type/uri.rb

```diff
@@ -0,0 +1,11 @@
+format do
+  include Phrase::Format
+
+  view :core do |args|
+    build_link _render_raw(args), render_title(args) || card.name
+  end
+
+  view :uri do |args|
+    build_link _render_raw(args)
```

Comment:
Normal types:

url view (already exists): http://wagnsite.com/cardname
uri_link view (proposed): <a href="http://wagnsite.com/cardname">
http://wagnsite.com/cardname</a>

(use case = when you're trying to get other people to promote a link to
your card.  this happens)

~~~~~

URI type:

url view: (same as raw)
uri_link view: <a href="(raw)">(raw)</a>


~~~~~~~~

If you zoom out, you'll see that it definitely makes sense.  Effectively
what you're saying is that nobody is really interested in linking to a URI
card; what they want to do is link to the resource it links to.  So in
cases where you might normally provide a card url, provide the destination
url instead.


On Thu, Jan 1, 2015 at 4:44 PM, Gerry Gleason <notifications@github.com>
wrote:

> In mod/05_standard/set/type/uri.rb
> <https://github.com/wagn/wagn/pull/281#discussion-diff-22403114>:
>
> > @@ -0,0 +1,11 @@
> > +format do
> > +  include Phrase::Format
> > +
> > +  view :core do |args|
> > +    build_link _render_raw(args), render_title(args) || card.name
> > +  end
> > +
> > +  view :uri do |args|
> > +    build_link _render_raw(args)
>
> It doesn't have any meaning for non-uri types. If you think otherwise,
> tell me what it would do? What would the link be targeted at? This is what
> I'm saying. The core view for most types is not a link, but it is for this
> type. So, the url_link view is a link for this type because the target is
> the raw content of my card. exactly like the core view, it's just that I
> also want that content in the element content. There isn't really an
> equivalent to this for other types. Core may not be right, maybe raw, but I
> don't see what other option would give something that makes sense.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/wagn/wagn/pull/281/files#r22403114>.
>



-- 
Ethan McCutchen
One of the Wagneers, Wagn.org

Wagn. How pioneers roll.

s: ethan.mccutchen
t: @intogreater